test/base/trainopts.js

```diff
@@ -173,3 +173,98 @@ describe('async train() options', () => {
       });
   }).timeout(10000);
 });
+
+describe('training options validation', () => {
+  it('iterations validation', () => {
+    let net = new brain.NeuralNetwork();
+    assert.throws(() => { net._updateTrainingOptions({ iterations: 'should be a string' }) });
+    assert.throws(() => { net._updateTrainingOptions({ iterations: () => {} }) });
+    assert.throws(() => { net._updateTrainingOptions({ iterations: false }) });
+    assert.throws(() => { net._updateTrainingOptions({ iterations: -1 }) });
+    assert.doesNotThrow(() => { net._updateTrainingOptions({ iterations: 5000 }) });
+  });
+
+  it('errorThresh validation', () => {
+    let net = new brain.NeuralNetwork();
+    assert.throws(() => { net._updateTrainingOptions({ errorThresh: 'no strings'}) });
+    assert.throws(() => { net._updateTrainingOptions({ errorThresh: () => {} }) });
+    assert.throws(() => { net._updateTrainingOptions({ errorThresh: 5}) });
+    assert.throws(() => { net._updateTrainingOptions({ errorThresh: -1}) });
+    assert.throws(() => { net._updateTrainingOptions({ errorThresh: false}) });
+    assert.doesNotThrow(() => { net._updateTrainingOptions({ errorThresh: 0.008}) });
+  });
+
+  it('log validation', () => {
+    let net = new brain.NeuralNetwork();
+    assert.throws(() => { net._updateTrainingOptions({ log: 'no strings' }) });
+    assert.throws(() => { net._updateTrainingOptions({ log: 4 }) });
+    assert.doesNotThrow(() => { net._updateTrainingOptions({ log: false }) });
+    assert.doesNotThrow(() => { net._updateTrainingOptions({ log: () => {} }) });
```

Comment:
is this a "valid" log function?

Reply:
yep, it would be user defined.  (logging allows the user to either set true and it will console.log or set their own method

Reply:
I don't understand why the `log` value can be user defined, that's the point of `callback`. Buuuuuut, I was a bit too hasty with this and the other "Valid callback" comment. This is fine with me.

Reply:
No worries.  The main idea is so that if the user has their own logging system (i.e. it writes to file or whatever) then they can keep the log to do just that, without having to use up their callback for that.

Reply:
Meh, I don't think it would "use up" their callback. But this is a moot point, we can continue discussion outside of this PR.